ProjFS.Mac/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -214,14 +214,21 @@ static VirtualizationRootHandle FindOrDetectRootAtVnode(vnode_t _Nonnull vnode,
         {
             // TODO: check xattr contents
             
-            char path[PrjFSMaxPath] = "";
-            int pathLength = sizeof(path);
-            errno_t error = vn_getpath(vnode, path, &pathLength);
+            const char* path = nullptr;
+#if DEBUG // Offline roots shouldn't need their path filled, and vn_getpath() may fail anyway.
```

Comment:
Why is this code in a Debug block now? If we don't need to fill paths for offline roots, can we just delete this code instead of only running it in Debug?

Reply:
I've found it helpful for diagnostics to be able to print out information relating to the root when doing development/debugging. Otherwise you've only got an index to go on, and an unretained vnode pointer. (which is not safe to pass to `KextLog_File*()`)

Reply:
@nickgra @pmj if we want to keep this around for debugging, what would you think of prepending path with something like "DEBUG:"?

That if something *did* accidentally take a dependency on having the offline root path we'd see it break when testing debug builds?

Reply:
@wilbaker sorry, I don't understand your suggestion, can you please elaborate?

Reply:
I think I see what you mean. We can remove the struct member outside of DEBUG builds entirely once the kext no longer needs to deal with relative paths. Do you think we'll still need to poison the string value in that case?

Reply:
> I think I see what you mean. We can remove the struct member outside of DEBUG builds entirely once the kext no longer needs to deal with relative paths. Do you think we'll still need to poison the string value in that case?

I think if we can remove the member completely outside of debug builds then we probably don't need to poison the value.  

I'm thinking we shouldn't be adding so much complexity in `#if DEBUG` that we take a dependency on it for behaving properly.

Reply:
> We can remove the struct member outside of DEBUG builds entirely once the kext no longer needs to deal with relative paths. Do you think we'll still need to poison the string value in that case?

Since the struct still has `path`, can we add the code to poison the path?  I checked and both the Mac functional tests and the Mac large repo build test run against the debug build of the kext.  

If a dependency on `path` is added I'm not sure that we'd catch it with out current tests.

Reply:
Yep, if we're not testing release builds, then that does sound essential. I've added that in commit 31a854f now.